internal/services/eventgrid/event_subscription_schema.go

```diff
@@ -124,7 +124,7 @@ func eventSubscriptionSchemaAzureFunctionEndpoint(conflictsWith []string) *plugi
 				"function_id": {
 					Type:         pluginsdk.TypeString,
 					Required:     true,
-					ValidateFunc: azure.ValidateResourceID, // TODO: validation for a Function App ID
+					ValidateFunc: webapps.ValidateFunctionID,
```

Comment:
Since this is technically changing what values are allowed for this property, we should wrap this in a feature flag for 5.0. While unlikely users have an incompatible ID in their config, out of an abundance of caution we should ensure this is only enabled once 5.0 is released.

(Can probably just be added to the other PR instead of this one)

```suggestion
					ValidateFunc: azure.ValidateResourceID, // TODO: validation for a Function App ID
```